samples/nrf9160/http_application_update/src/main.c

```diff
@@ -100,7 +101,47 @@ static int dfu_button_init(void)
 	return 0;
 }
 
+/**@brief Callback handler for DFU target events
+ *
+ * Turns the modem into offline mode to handle the current pending erase request
+ * so that the modem will have time to service the erase request.
+ */
+static void dfu_target_callback_handler(enum dfu_target_evt_id evt)
+{
+	int err;
+	enum lte_lc_func_mode mode;
 
+	err = lte_lc_func_mode_get(&mode);
+	if (err < 0) {
+		__ASSERT(err == 0, "Unable to get LTE mode");
+	}
+
+	switch (evt) {
+	case DFU_TARGET_EVT_TIMEOUT:
+		printk("Modem firmware bank erase pending\n");
+		if (mode == LTE_LC_FUNC_MODE_POWER_OFF) {
+			break;
+		} else if (mode == LTE_LC_FUNC_MODE_NORMAL) {
```

Comment:
It can timeout if you set your timeout short enough. The idea is to create a general callback function which the customers can look at. So that it can be called before and after you've connected to the LTE network hence the early return.

Reply:
One could introduce a dependency in `dfu_target_modem` that checks the `lte_lc_func` mode and if it is not connected ignore the time out if that's cleaner.